File: POO/POOHeranca/app/src/main/java/com/example/pooheranca/MainActivity.java
package com.example.pooheranca;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        Funcionario joao = new Funcionario();
        Funcionario pedro = new Funcionario();
        Gerente gerentePedro = new Gerente();

        gerentePedro.setNome("pedro");
        gerentePedro.setDataAdmissao("01/02/2024");
        gerentePedro.setSalario(2000.50);
        gerentePedro.setComissao(15);
        gerentePedro.setTotalFuncionarios(10);
        gerentePedro.setDataNascimento("01/02/1999");
    }
}

File: POOHerancaKotlin/app/src/main/java/com/example/pooherancakotlin/MainActivity.kt
package com.example.pooherancakotlin

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        lateinit var joao: Funcionario

        joao.setNome("João")
        joao.setSalario(1200)
        joao.setDataAdmissao("01/01/2024")

        lateinit var gerentePedro: Gerente

        gerentePedro.setNome("Pedro")
        gerentePedro.setDataAdmissao("01/01/2022")
        gerentePedro.setSalario(2400)
        gerentePedro.setComissao(10.5)
        gerentePedro.setTotalFuncionarios(25)


    }
}